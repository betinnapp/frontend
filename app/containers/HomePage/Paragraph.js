import styled from 'styled-components'

const Paragraph = styled.p`
  line-height: 60px;
  margin: 0;
  font-weight: ${props => props.bold && 'bold'};
  font-size: ${props => (props.fontSize && `${props.fontSize}px`) || '1em'};
`

export default Paragraph
