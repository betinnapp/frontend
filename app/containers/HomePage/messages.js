/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl'

export const scope = 'app.containers.HomePage'

export default defineMessages({
  learnToManagerYourFinances: {
    id: `${scope}.learnToManagerYourFinances`,
    defaultMessage: 'Aprenda a gerenciar suas finanças de graça.',
  },
  startNow: {
    id: `${scope}.startNow`,
    defaultMessage: 'Comece agora',
  },
})
