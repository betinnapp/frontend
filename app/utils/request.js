import axios from 'axios'

export default function request(url, options) {
  const newOptions = {
    url,
    ...options,
  }

  if (!options.headers) {
    newOptions.headers = {
      'Content-type': 'application/json',
    }
  }

  return axios(newOptions).then(response => response.data)
}
