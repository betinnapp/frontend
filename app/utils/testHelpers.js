import React from 'react'
import { render } from 'react-testing-library'
import { IntlProvider } from 'react-intl'

import ptTranslations from 'translations/pt.json'

export function renderWithIntl(component) {
  return render(
    <IntlProvider locale="pt" messages={ptTranslations}>
      {component}
    </IntlProvider>,
  )
}
