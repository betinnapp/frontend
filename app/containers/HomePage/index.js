/**
 *
 * HomePage
 *
 */

import React from 'react'
import { FormattedMessage } from 'react-intl'
import styled from 'styled-components'

import Button from 'components/Button'
import Link from 'components/Link'
import Slogan from 'components/Slogan'
import stat from 'images/stat.svg'

import messages from './messages'
import Paragraph from './Paragraph'

const Wrapper = styled.div`
  display: grid;
  grid-template-rows: 1fr 2fr 1fr;
  align-items: center;
  justify-content: center;
  justify-items: center;
  text-align: center;
  height: 100%;

  > *:first-child {
    justify-self: start;
  }
`

export default function HomePage() {
  const startNowOnClickHandler = () => {}

  return (
    <Wrapper>
      <Slogan />
      <div>
        <img src={stat} alt="Stats" />
        <Paragraph bold fontSize="18">
          <FormattedMessage {...messages.learnToManagerYourFinances} />
        </Paragraph>
      </div>
      <div>
        <Button id="startNow" onClick={startNowOnClickHandler}>
          <FormattedMessage {...messages.startNow} />
        </Button>
        <Paragraph>
          <Link id="haveAnAccount" to="/login">
            <FormattedMessage {...messages.alreadyHaveAnAccount} />
          </Link>
        </Paragraph>
      </div>
    </Wrapper>
  )
}
