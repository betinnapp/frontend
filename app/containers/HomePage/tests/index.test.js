import React from 'react'
import { renderWithIntl } from 'utils/testHelpers'

import HomePage from '../index'

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderWithIntl(<HomePage />)
    expect(firstChild).toMatchSnapshot()
  })
})
