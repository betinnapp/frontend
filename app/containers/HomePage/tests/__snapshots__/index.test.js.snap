// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <img
    alt="Stats"
    src="IMAGE_MOCK"
  />
  <span>
    Aprenda a gerenciar suas finanças de graça.
  </span>
  <button
    class="Button__StyledButton-ss41oc-0 eUuPsm"
    id="startNow"
    type="button"
  >
    <span>
      Comece agora
    </span>
  </button>
</div>
`;
